src/components/encoding-pane/property-editor.tsx

```diff
@@ -27,21 +28,24 @@ export class PropertyEditorBase extends React.PureComponent<PropertyEditorProps,
 
   public render() {
     const {prop, nestedProp, propTab, shelfId, fieldDef} = this.props;
-    const {schema, uiSchema} = generatePropertyEditorSchema(prop, nestedProp, propTab, fieldDef, shelfId);
-    const formData = generateFormData(shelfId, fieldDef);
-    return (
-      <div styleName="property-editor">
-        <Form
-          schema={schema}
-          uiSchema={uiSchema}
-          formData={formData}
-          onChange={this.changeFieldProperty}
-        >
-          <button type="submit" style={{display: 'none'}}>Submit</button>
-          {/* hide required submit button */}
-        </Form>
-      </div>
-    );
+    if (!isWildcardChannelId(shelfId)) {
```

Comment:
`const {channel} = shelfId;` then apply type guard to channel so you don't have to cast.  

(As said, avoid casting at all cost unless it's really super annoying.)